docs/adr/03-errrors.md

```diff
@@ -0,0 +1,59 @@
+# Errors
+
+Returning meaningful errors is important for users of our libraries. The Rust bitcoin ecosystem uses descriptive and data-driven errors, and we would like to stay as close to them as possible when building language bindings for these libraries.
+
+## Context and Problem Statement
+
+The tool we use to produce language bindings for bdk-ffi libraries is [uniffi]. While the library is powerful, it also comes with some caveats. Those come into play when attempting to expose Rust errors; we must choose between simple enums that have variants but not data associated with them (for example a `TxidError` would not be able to return the specific txid that triggered the error), or more complex objects that do have data fields, but no ability to control the error message returned to the user. What's more, while Kotlin and Java users expect a `message` field on an `Exception` type, this field becomes an empty string if the object has no data, and just a strignified version of the fields if it does have any.
```

Comment:
if you want to touch on swift real quick here we can add something like this at the end of the `Context and Problem Statement` section:

"Swift, in contrast, can optionally leverage something like the `LocalizedError` protocol to provide custom and localized descriptions."